src/commands/inputs.rs

```diff
@@ -0,0 +1,279 @@
+                self.literal_defaults.insert(name, has_literal_default);
+            }
+        }
+    }
+}
+
+// --hide-expressions hides any input which isn't defaulted to a literal or
+// required
+
+pub async fn generate_inputs(args: InputsArgs) -> Result<()> {
+    let results: Vec<wdl::analysis::AnalysisResult> =
+        analyze(args.document.as_str(), vec![], false, false).await?;
+
+    let uri: Url = Url::parse(args.document.as_str()).unwrap_or_else(|_| {
+        path_to_uri(args.document.as_str()).expect("file should be a local path")
+    });
+
+    let result = results
+        .iter()
+        .find(|r| **r.document().uri() == uri)
+        .context("failed to find document in analysis results")?;
+
+    let document: &std::sync::Arc<Document> = result.document();
+
+    let diagnostics: &[Diagnostic] = document.diagnostics();
+    if diagnostics
+        .iter()
+        .any(|d| d.severity() == wdl::ast::Severity::Error)
+    {
+        anyhow::bail!("Failed to parse WDL document: {:?}", diagnostics);
+    }
+
+    let mut template = serde_json::Map::new();
+
+    // Collect inputs and their parent information
+    let inputs_with_parents = collect_inputs_with_parents(&args, document)?;
+
+    for (parent_name, name, input) in inputs_with_parents {
+        // Skip if hide_defaults is true and input has any default
+        if args.hide_defaults && has_default(document, args.document.as_str(), name) {
+            continue;
+        }
+
+        // Skip if hide_expressions is true and input has non-literal default
+        if args.hide_expressions && !has_literal_default(document, args.document.as_str(), name) {
+            continue;
+        }
+
+        let v: &wdl::analysis::types::Type = input.ty();
+        let key = format!("{}.{}", parent_name, name);
+        let value = type_to_json(&v);
+        template.insert(key, value);
+    }
+
+    let json_output = serde_json::to_string_pretty(&template)?;
+
+    if let Some(output_path) = args.output {
+        std::fs::write(output_path, json_output)?;
+    } else {
+        println!("{}", json_output);
+    }
+
+    Ok(())
+}
+
+fn type_to_json(ty: &Type) -> Value {
+    match ty {
+        Type::Primitive(ty, _bool) => match ty {
+            wdl::analysis::types::PrimitiveType::Boolean => Value::String("Boolean".to_string()),
+            wdl::analysis::types::PrimitiveType::Integer => Value::String("Integer".to_string()),
+            wdl::analysis::types::PrimitiveType::Float => Value::String("Float".to_string()),
+            wdl::analysis::types::PrimitiveType::String => Value::String("String".to_string()),
+            wdl::analysis::types::PrimitiveType::File => Value::String("File".to_string()),
+            wdl::analysis::types::PrimitiveType::Directory => {
+                Value::String("Directory".to_string())
+            }
+        },
+
+        _ => Value::Null,
+    }
+}
+
+fn has_default(document: &Document, document_path: &str, input_name: &str) -> bool {
```

Comment:
We don't only care if the input has a default, but also what that default is. You will need to update this to extract the expression as well so that we can render that in the output JSON.